MIDAS/src/rocket_state.h

```diff
@@ -1,49 +1,94 @@
 #pragma once
 
+#include <array>
+
+#include "finite-state-machines/fsm.h"
+
 #include "sensor_data.h"
 #include "hal.h"
+#include "Buffer.h"
+
+
 
 /** The RocketState struct stores everything that is needed by more than one system/thread of the Rocket.
  *
  *  Normally, this would be considered a poor decision. However, the fact that all this data is here
  *  makes it easier to debug since all this data can be logged (and thus used when debugging).
  */
 
-template<typename SensorData>
+template<typename S>
 struct SensorState {
 private:
-    Mutex<SensorData> current;
-    Queue<SensorData> queue;
+    Mutex<S> current;
+    Queue<S> queue;
 
 public:
-    void update(SensorData data) {
+    void update(S data) {
         current.write(data);
         queue.send(data);
     };
 
-    SensorData getRecent() {
+    S getRecent() {
         return current.read();
     };
 
-    bool getQueued(SensorData* out) {
+    bool getQueued(S* out) {
         return queue.receive(out);
     };
 
-    SensorState() : current(SensorData()) { }
+    SensorState() : current(S()) { }
+};
+
+template<typename S, size_t count>
+struct BufferedSensorState {
+private:
+    Mutex<S> current;
+    Queue<S> queue;
+    Buffer<S, count> buffer;
+
+public:
+    void update(S data) {
+        current.write(data);
+        queue.send(data);
+        buffer.push(data);
+    };
+
+    S getRecent() {
+        return current.read();
+    };
+
+    // wrapper function to get easy access to buffer data
+    std::array<S, count> getBufferRecent() {
+        std::array<S, count> arr = buffer. template read_recent<count>(); 
+        return arr;
+    };
+    
+    bool getQueued(S* out) {
+        return queue.receive(out);
+    };
+
+    BufferedSensorState() : current(S()) { }
+};
+
+enum stage {
+    sustainer = true,
+    booster = false
```

Comment:
if u have bools, always name them as `is_<...>` but also why r we assigning bools to an enum